TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Workspace.php

```diff
@@ -461,17 +461,17 @@ protected function replaceNodeData(NodeInterface $node, NodeData $targetNodeData
             }
         }
 
-        if ($node->isRemoved() === true) {
-            $this->nodeDataRepository->remove($targetNodeData);
-        } else {
-            $targetNodeData->similarize($node->getNodeData());
-            if ($nodeWasMoved) {
-                $targetNodeData->setPath($node->getPath(), false);
-            }
-            $targetNodeData->setLastPublicationDateTime($this->now);
-            $node->setNodeData($targetNodeData);
-            $this->nodeService->cleanUpProperties($node);
+        $targetNodeData->setRemoved($node->isRemoved());
+        $targetNodeData->similarize($sourceNodeData);
+        // TODO: This seemswrong and introduces a publish order between nodes. We should always set the path.
```

Comment:
The behavior with this check is:

Live node /a/b/c

User X moves that node to /a/b
User Y changes a property in /a/b/c

now afterwards user X publishes, then user Y publishes.
The result now is a node /a/b with changed property.

If first Y then X publishes you would have a node /a/b withOUT changed property.

If we always set the path the result would be:
X first then Y: Node /a/b/c with changed property
Y first then X: Node /a/b withOUT changed property

That's way more reliable as you always get what you saw in your workspace when publishing. The current behavior is a bad mix IMHO.